### PR TITLE
[WIP] fix use of AnsibleError for inventory plugins

### PR DESCRIPTION
**Pull request by s-hertel**
##### SUMMARY
Use AnsibleError for fatal errors that should exclude a source from being parsed by any inventory. Use AnsibleParserError for non-fatal errors so other plugins can attempt to parse.

needs:
- [ ] inventory/manager.py fix
- [ ] docs/porting guide

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
inventory plugins

### CODE REVIEW

File: lib/ansible/plugins/inventory/linode.py [hunk 1]
```diff
@@ -193,6 +181,9 @@ def verify_file(self, path):
 
     def parse(self, inventory, loader, path, cache=True):
         """Dynamically parse Linode the cloud inventory."""
+        if not HAS_LINODE_API4:
+            raise AnsibleParserError('the Linode dynamic inventory plugin requires linode_api4.')
```
Comment (decentral1se) [inline lib/ansible/plugins/inventory/linode.py]: Oh why not :sweat_smile: 

```suggestion
            raise AnsibleParserError('The Linode dynamic inventory plugin requires linode_api4.')
```

File: lib/ansible/plugins/inventory/linode.py [hunk 2]
```diff
@@ -76,27 +77,14 @@ def _build_client(self):
         """Build the Linode client."""
 
         access_token = self.get_option('access_token')
-
-        if access_token is None:
```
Comment (decentral1se) [inline lib/ansible/plugins/inventory/linode.py:80]: This conditional was removed but not moved somewhere else?

Comment (s-hertel) [reply lib/ansible/plugins/inventory/linode.py:80]: This should be handled in the documentation, which doubles as an argument specification for inventory plugins. The access_token is listed as required, and it falls back to the LINODE_ACCESS_TOKEN environment variable. If the option isn't provided, the plugin should fail before it ever gets to this code path.

File: lib/ansible/plugins/inventory/linode.py (hunk 1 above)
Comment (s-hertel) [reply lib/ansible/plugins/inventory/linode.py]: Heh, just a copy-pasta from what it was before :-)

**Comment by s-hertel** [comment]
I'll reopen this when I have time to make it not-a-WIP and add documentation. I'm sure there are more misuses since I opened this and some thought needs to go into 3rd party plugins as well, probably.

**Pull request closed without merging**